app/src/main/java/com/github/drawyourpath/bootcamp/path/Path.kt

```diff
@@ -0,0 +1,89 @@
+package com.github.drawyourpath.bootcamp.path
+import com.google.type.Color
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.Polyline
+import com.google.android.gms.maps.model.PolylineOptions
+
+// Define a Path class that represents a runner's path.
+class Path {
+
+    // Constructor that creates an empty path.
+    constructor()
+
+    // Constructor that creates a path from a list of points.
+    constructor(points: List<LatLng>) {
+        this.points.addAll(points)
+    }
+
+    //private mutable list to store the path as a collection of points.
+    private val points = mutableListOf<LatLng>()
+
+    // Add a point to the path's list of points(at the end of the list).
+    fun addPoint(point: LatLng) {
+        points.add(point)
+    }
+
+    // Return an immutable list of the path's points.
+    fun getPoints(): List<LatLng> {
+        return points.toList()
+    }
+
+    // Clear the path's list of points.
+    fun clear() {
+        points.clear()
+    }
+
+    // Return the number of points in the path.
+    fun size(): Int {
+        return points.size
+    }
+
+    //helper function to calculate distance between two points(in meters)
+    private fun distance(point1: LatLng, point2: LatLng): Double {
+        val lat1 = point1.latitude
+        val lat2 = point2.latitude
+        val lon1 = point1.longitude
+        val lon2 = point2.longitude
+        val R = 6371e3 // metres
+        val φ1 = lat1 * Math.PI / 180 // φ, λ in radians

```

Comment:
Try to to not used mathematical characters like φ or λ for the code style implementation.